corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -349,6 +349,9 @@ def _extract_case_blocks(data, path=None):
 
     data must be json representing a node in an xform submission
     """
+    from corehq.form_processor.utils import extract_meta_instance_id
+    form_id = extract_meta_instance_id(data)
```

Comment:
I don't think this'll quite work, because `_extract_case_blocks` is recursive (see call a few lines down). Thus, this would only work for the outermost data. Another option is to pass `form_id` _into_ `_extract_case_blocks` from `extract_case_blocks`, where (to my understanding) it should always be available.